core/logging.py

```diff
@@ -52,27 +46,95 @@ def emit(self, record: logging.LogRecord):
         )
 
 
-oldLevels = {k: v.getEffectiveLevel()
-             for k, v in logging.root.manager.loggerDict.items()
-             if isinstance(v, logging.Logger)}
+# ViURLocalFormatter ---------------------------------------------------------------------------------------------------
+
+class ViURLocalFormatter(logging.Formatter):
+    """
+    This is a formatter that injects console color sequences for debug output.
+    """
+
+    BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE = range(8)
+    RESET_SEQ = "\033[0m"
+    COLOR_SEQ = "\033[1;%dm"
+    BOLD_SEQ = "\033[1m"
+
+    LEVELS = {
+        "WARNING": YELLOW,
+        "INFO": CYAN,
+        "DEBUG": BLUE,
+        "CRITICAL": RED,
+        "ERROR": RED
+    }
+
+    def format(self, record):
+        if "pathname" in record.__dict__.keys():
+            # truncate the pathname
+            if "/deploy" in record.pathname:
+                pathname = record.pathname.split("/deploy/")[1]
+            else:
+                pathname = record.pathname
+                if len(pathname) > 20:
+                    pathname = ".../" + "/".join(pathname.split("/")[-3:])
+
+            record.pathname = pathname
+
+        levelname = record.levelname
+
+        if levelname in ViURLocalFormatter.LEVELS:
+            record.levelname = ViURLocalFormatter.COLOR_SEQ % (30 + ViURLocalFormatter.LEVELS[levelname]) \
+                               + levelname + ViURLocalFormatter.RESET_SEQ
+
+        return super().format(record)
+
+
+# Logger config
+
+client = google.cloud.logging.Client()
+requestLogger = client.logger("ViUR")
+requestLoggingRessource = Resource(
+    type="gae_app",
+    labels={
+       "project_id": conf["viur.instance.project_id"],
+       "module_id": "default",
+       "version_id": conf["viur.instance.app_version"] if not conf[
+           "viur.instance.is_dev_server"] else "dev_appserver",
+    }
+)
 
 logger = logging.getLogger()
 logger.setLevel(logging.DEBUG)
 
-for loggerName, level in oldLevels.items():
-    logging.getLogger(loggerName).setLevel(level)
+# Inherit old logger levels
+for name, level in {
+            k: v.getEffectiveLevel()
+            for k, v in logging.root.manager.loggerDict.items()
+            if isinstance(v, logging.Logger)
+        }.items():
+    logging.getLogger(name).setLevel(level)
 
+# Remove handlers from logger
 for handler in logger.handlers[:]:
     logger.removeHandler(handler)
 
-handler = ViURDefaultLogger(client, name="ViUR-Messages", resource=Resource(type="gae_app", labels={}))
-logger.addHandler(handler)
+# Plug-in ViURDefaultLogger & custom formatter when running in the cloud
+if not conf["viur.instance.is_dev_server"]:
+    handler = ViURDefaultLogger(client, name="ViUR-Messages", resource=Resource(type="gae_app", labels={}))
+    logger.addHandler(handler)
+
+    formatter = logging.Formatter("%(levelname)-8s %(asctime)s %(filename)s:%(lineno)s] %(message)s")
```

Comment:
I think we don't need a `StreamHandler` in the cloud. So this should be not necessary. Then you can add an indent to the lines 133-135 (and move them into the local dev appserver block):
https://github.com/viur-framework/viur-core/blob/1d762d9abf071e8f5794529118c855ad7eb92bb8/core/logging.py#L133-L135

Reply:
The StreamHandler was moved into the local debug block